packages/netsuite-adapter/src/config.ts

```diff
@@ -179,6 +219,76 @@ const clientConfigType = createMatchingObjectType<SdfClientConfig>({
   },
 })
 
+const validateInstalledSuiteApps = (installedSuiteApps: unknown): void => {
+  validateArrayOfStrings(installedSuiteApps, [CONFIG.client, CLIENT_CONFIG.installedSuiteApps])
+  const invalidValues = installedSuiteApps.filter(id => !SUITEAPP_ID_FORMAT_REGEX.test(id))
+  if (invalidValues.length !== 0) {
+    throw new Error(`${CLIENT_CONFIG.installedSuiteApps} values should contain only lowercase characters or numbers and exactly two dots (such as com.saltoio.salto). The following values are invalid: ${invalidValues.join(', ')}`)
+  }
+}
+
+const isCustomRecordTypeName = (name: string): boolean => name.startsWith(CUSTOM_RECORD_TYPE_NAME_PREFIX)
+
+function validateMaxInstancesPerType(maxInstancesPerType: unknown):
+  asserts maxInstancesPerType is MaxInstancesPerType[] {
+  if (Array.isArray(maxInstancesPerType) && maxInstancesPerType.every(
+    val => 'name' in val && 'limit' in val && typeof val.name === 'string' && typeof val.limit === 'number'
+  )) {
+    const invalidTypes = maxInstancesPerType.filter(maxType =>
+      !regex.isValidRegex(maxType.name)
+      || (regex.isValidRegex(maxType.name)
+        && noSupportedTypeMatch(maxType.name)
+        && !isCustomRecordTypeName(maxType.name)))
+    if (invalidTypes.length > 0) {
+      throw new Error(
+        `The following types or regular expressions in ${CLIENT_CONFIG.maxInstancesPerType}`
+        + ` do not match any supported type: ${safeJsonStringify(invalidTypes)}`
+      )
+    }
+  } else {
+    throw new Error(`Expected ${CLIENT_CONFIG.maxInstancesPerType} to be a list of { name: string, limit: number },`
+    + ` but found:\n${safeJsonStringify(maxInstancesPerType, undefined, 4)}.`)
+  }
+}
+
+export function validateClientConfig(
+  client: unknown,
+  fetchTargetDefined: boolean
+): asserts client is SdfClientConfig {
+  validatePlainObject(client, CONFIG.client)
+  const {
+    fetchAllTypesAtOnce,
+    installedSuiteApps,
+    maxInstancesPerType,
+  } = _.pick(client, Object.values(CLIENT_CONFIG))
```

Comment:
nit - since we check `validatePlainObject` before calling `validateClientConfig`:
```suggestion
export function validateClientConfig(
  client: Record<string, unknown>,
  fetchTargetDefined: boolean
): asserts client is SdfClientConfig {
  const {
    fetchAllTypesAtOnce,
    installedSuiteApps,
    maxInstancesPerType,
  } = _.pick(client, Object.values(CLIENT_CONFIG))
```